kernel/src/net/socket/ip/datagram/bound.rs

```diff
@@ -38,12 +39,23 @@ impl BoundDatagram {
         self.remote_endpoint = Some(*endpoint)
     }
 
-    pub fn try_recv(&self, buf: &mut [u8], _flags: SendRecvFlags) -> Result<(usize, IpEndpoint)> {
-        let result = self
-            .bound_socket
-            .raw_with(|socket: &mut RawUdpSocket| socket.recv_slice(buf));
+    pub fn try_recv(
+        &self,
+        writers: &mut dyn IoVecWrite,
+        _flags: SendRecvFlags,
+    ) -> Result<(usize, IpEndpoint)> {
+        let result = self.bound_socket.raw_with(|socket: &mut RawUdpSocket| {
+            socket.recv().map(|(packet, udp_metadata)| {
+                println!("packet = {:?}", packet);
+                let copied_res = writers.write_from(&mut VmReader::from(packet));
+                println!("copied_res = {:?}", copied_res);
+                let endpoint = udp_metadata.endpoint;
+                (copied_res, endpoint)
+            })
+        });
         match result {
```

Comment:
Add a blank line before `match result` to be a bit more consistent with TCP's `try_send`/`try_recv` implementation,